### PR TITLE
Enable support for nint/nuint for Vector64/128/256<T>

### LINKED ISSUE #52017: Extend Vector64<T>, Vector128<T>, and Vector256<T> to support nint and nuint
### Proposal
Extend `Vector64<T>`, `Vector128<T>`, and `Vector256<T>` to support `nint` and `nuint` as valid primitive types. This will extend a number of existing generic functions which take a `Vector<T>` to also support taking the new types rather than throwing a `PlatformNotSupportedException`.

Additionally, the following non-generic APIs should be added for parity with the existing surface area:
```csharp
namespace System.Runtime.Intrinsics
{
    public static partial class Vector64
    {
        public static Vector64<nint> AsNInt<T>(Vector64<T> value);
        public static Vector64<nuint> AsNUInt<T>(Vector64<T> value);

        public static Vector64<nint> Create(nint value);
        public static Vector64<nuint> Create(nuint value);

        public static Vector64<nint> CreateScalar(nint value);
        public static Vector64<nuint> CreateScalar(nuint value);

        public static Vector64<nint> CreateScalarUnsafe(nint value);
        public static Vector64<nuint> CreateScalarUnsafe(nuint value);
    }

    public static partial class Vector128
    {
        public static Vector128<nint> AsNInt<T>(Vector128<T> value);
        public static Vector128<nuint> AsNUInt<T>(Vector128<T> value);

        public static Vector128<nint> Create(nint value);
        public static Vector128<nuint> Create(nuint value);

        public static Vector128<nint> Create(Vector64<nint> lower, Vector64<nint> upper);
        public static Vector128<nuint> Create(Vector64<nuint> lower, Vector64<nuint> upper);

        public static Vector128<nint> CreateScalar(nint value);
        public static Vector128<nuint> CreateScalar(nuint value);

        public static Vector128<nint> CreateScalarUnsafe(nint value);
        public static Vector128<nuint> CreateScalarUnsafe(nuint value);
    }

    public static partial class Vector256
    {
        public static Vector256<nint> AsNInt<T>(Vector256<T> value);
        public static Vector256<nuint> AsNUInt<T>(Vector256<T> value);

        public static Vector256<nint> Create(nint value);
        public static Vector256<nuint> Create(nuint value);

        public static Vector256<nint> Create(Vector128<nint> lower, Vector128<nint> upper);
        public static Vector256<nuint> Create(Vector128<nuint> lower, Vector128<nuint> upper);

        public static Vector256<nint> CreateScalar(nint value);
        public static Vector256<nuint> CreateScalar(nuint value);

        public static Vector256<nint> CreateScalarUnsafe(nint value);
        public static Vector256<nuint> CreateScalarUnsafe(nuint value);
    }
}
```

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves https://github.com/dotnet/runtime/issues/52017

### CODE REVIEW

**Comment by tannergooding** [comment]
The bulk of the change here is removing the `NotSupported` tests for `nint`/`nuint`

File: src/coreclr/jit/hwintrinsic.cpp [hunk 1]
```diff
@@ -596,12 +596,6 @@ static bool isSupportedBaseType(NamedIntrinsic intrinsic, CorInfoType baseJitTyp
         return false;
     }
 
-    if ((baseJitType == CORINFO_TYPE_NATIVEINT) || (baseJitType == CORINFO_TYPE_NATIVEUINT))
-    {
-        // We don't want to support the general purpose helpers for nint/nuint until after they go through API review.
-        return false;
-    }
-
```
Comment (tannergooding) [inline src/coreclr/jit/hwintrinsic.cpp:604]: This is the JIT change. Just had to remove the if statement blocking the types.

**Review by echesakov — APPROVED** [review]
The JIT change looks good.

File: src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs [hunk 2]
```diff
@@ -1053,6 +1125,7 @@ static Vector64<uint> SoftwareFallback(uint e0, uint e1)
         /// <summary>Creates a new <see cref="Vector64{Byte}" /> instance with the first element initialized to the specified value and the remaining elements initialized to zero.</summary>
         /// <param name="value">The value that element 0 will be initialized to.</param>
         /// <returns>A new <see cref="Vector64{Byte}" /> instance with the first element initialized to <paramref name="value"/> and the remaining elements initialized to zero.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```
Comment (GrabYourPitchforks) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs:1128]: I assume these additions were intentional?

Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector64.cs:1128]: Yes, they already existed for Vector128/256 but were missed for Vector64 in a previous PR.

**Review by GrabYourPitchforks — APPROVED** [review]

**Pull request merged**